tools/SampleAnalyzer/Commons/DataFormat/MCEventFormat.h

```diff
@@ -37,206 +35,206 @@
+            // weight_ = 1.;
+            scale_ = 0.;
+            alphaQED_ = 0.;
+            alphaQCD_ = 0.;
+            TET_ = 0.;
+            THT_ = 0.;
+            Meff_ = 0.;
+        }
+
+        /// Destructor
+        ~MCEventFormat() {}
+
+        /// Accessor to the Missing Transverse Energy (read-only)
+        const MCParticleFormat &MET() const { return MET_; }
+
+        /// Accessor to the Missing Hadronic Transverse Energy (read-only)
+        const MCParticleFormat &MHT() const { return MHT_; }
+
+        /// Accessor to the Total Transverse Energy (read-only)
+        const MAfloat64 &TET() const { return TET_; }
+
+        /// Accessor to the Total Hadronic Transverse Energy (read-only)
+        const MAfloat64 &THT() const { return THT_; }
+
+        /// Accessor to the Total effective mass (read-only)
+        const MAfloat64 &Meff() const { return Meff_; }
+
+        /// Accessor to the Missing Transverse Energy
+        MCParticleFormat &MET() { return MET_; }
+
+        /// Accessor to the Missing Hadronic Transverse Energy
+        MCParticleFormat &MHT() { return MHT_; }
+
+        /// Accessor to the Total Transverse Energy
+        MAfloat64 &TET() { return TET_; }
+
+        /// Accessor to the Total Hadronic Transverse Energy
+        MAfloat64 &THT() { return THT_; }
+
+        /// Accessor to the Total effective mass
+        MAfloat64 &Meff() { return Meff_; }
+
+        /// Accessor to the process identity
+        const MAuint32 &processId() const { return processId_; }
+
+        /// Accessor to the event weight
+        const MAfloat64 &weight() const { return multiweights_.Get(0); }
+
+        /// Accessor to the scale
+        const MAfloat64 &scale() const { return scale_; }
+
+        /// Accessor to alpha_QED
+        const MAfloat64 &alphaQED() const { return alphaQED_; }
+
+        /// Accessor to alpha_QCD
+        const MAfloat64 &alphaQCD() const { return alphaQCD_; }
+
+        /// Accessor to multiweights
+        const WeightCollection &multiweights() const { return multiweights_; }
+
+        /// Accessor to multiweights
+        WeightCollection &weights() { return multiweights_; }
+
+        /// Accessor to multiweights
+        const WeightCollection &weights() const { return multiweights_; }
+
+        /// Accessor to multiweights
+        const MAfloat64 &get_weight(MAuint32 id) const { return multiweights_.Get(id); }
 
-  /// Displaying data 
-  void PrintVertices() const;
+        /// Accessor to the generated particle collection (read-only)
+        const std::vector<MCParticleFormat> &particles() const { return particles_; }
+
+        /// Accessor to the generated particle collection
+        std::vector<MCParticleFormat> &particles() { return particles_; }
+
+        /// Setting the process identity
+        void setProcessId(MAuint32 v) { processId_ = v; }
+
+        /// Setting the event weight
+        // void setWeight(MAfloat64 v) const { weight_ = v; }
```

Comment:
No more ways to set a weight?

Reply:
Not from here, it has to be through weight collection there is a ``Add`` function.

Reply:
Okay why not. Although the shortcut function may be good to have.

Reply:
Sure I can add a shortcut